huaweicloud/services/mpc/resource_huaweicloud_mpc_transcoding_template_group.go

```diff
@@ -186,224 +187,325 @@ func ResourceTranscodingTemplateGroup() *schema.Resource {
+		httpUrl = "v1/{project_id}/template_group/transcodings"
+	)
+
+	client, err := cfg.NewServiceClient("mpc", region)
 	if err != nil {
-		return diag.Errorf("error creating MPC client : %s", err)
+		return diag.Errorf("error creating MPC client: %s", err)
 	}
 
-	createOpts := mpc.TransTemplateGroup{
-		Name:        d.Get("name").(string),
-		Videos:      buildVideosOpts(d.Get("videos").([]interface{})),
-		VideoCommon: buildVideoCommonOpts(d.Get("video_common").([]interface{})),
-		Audio:       buildAudioOpts(d.Get("audio").([]interface{})),
-		Common:      buildCommonOpts(d),
+	createPath := client.Endpoint + httpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildCreateTemplateGroupBodyParams(d)),
 	}
 
-	createReq := mpc.CreateTemplateGroupRequest{
-		Body: &createOpts,
+	resp, err := client.Request("POST", createPath, &createOpt)
+	if err != nil {
+		return diag.Errorf("error creating MPC transcoding template group: %s", err)
 	}
 
-	resp, err := client.CreateTemplateGroup(&createReq)
+	respBody, err := utils.FlattenResponse(resp)
 	if err != nil {
-		return diag.Errorf("error creating MPC transcoding template group: %s", err)
+		return diag.FromErr(err)
 	}
 
-	d.SetId(*resp.TemplateGroup.GroupId)
+	groupId := utils.PathSearch("template_group.group_id", respBody, "").(string)
+	if groupId == "" {
+		return diag.Errorf("error creating MPC transcoding template group: ID is not found in API response")
+	}
+
+	d.SetId(groupId)
 
 	return resourceTranscodingTemplateGroupRead(ctx, d, meta)
 }
 
 func resourceTranscodingTemplateGroupRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	config := meta.(*config.Config)
-	client, err := config.HcMpcV1Client(config.GetRegion(d))
+	var (
+		cfg    = meta.(*config.Config)
+		region = cfg.GetRegion(d)
+	)
+
+	client, err := cfg.NewServiceClient("mpc", region)
 	if err != nil {
-		return diag.Errorf("error creating MPC client : %s", err)
+		return diag.Errorf("error creating MPC client: %s", err)
 	}
 
-	resp, err := client.ListTemplateGroup(&mpc.ListTemplateGroupRequest{GroupId: &[]string{d.Id()}})
+	respBody, err := GetTranscodingTemplateGroup(client, []string{d.Id()})
 	if err != nil {
-		return common.CheckDeletedDiag(d, err, "error retrieving MPC transcoding template group")
+		return common.CheckDeletedDiag(d, common.ConvertExpected403ErrInto404Err(err, "error_code", "MPC.10231"),
+			"error retrieving MPC transcoding template group")
 	}
 
-	templateGroupList := *resp.TemplateGroupList
+	mErr := multierror.Append(
+		d.Set("region", region),
+		d.Set("name", utils.PathSearch("name", respBody, nil)),
+		d.Set("videos", flattenTemplateGroupVideos(utils.PathSearch("videos", respBody, make([]interface{}, 0)))),
+		d.Set("audio", flattenTemplateGroupAudio(utils.PathSearch("audio", respBody, make([]interface{}, 0)))),
```

Comment:
The audio response is of type object. Why is the default value configured as an array?